app/main.py

```diff
@@ -98,6 +101,9 @@ def grab_secrets(self):
                    _logger.info('Retrieving secret name:%s with version: %s', key_name, key_version)
            	   secret = client.get_secret(vault_base_url, key_name, key_version)
                    output_path = os.path.join(secrets_output_folder, key_name)
+                   if secret.kid is not None:
+                         _logger.info('Secret is backing certificate. Dumping private key and certificate.')
+                         self.dump_pfx(secret.value, key_name, keys_output_folder, certs_output_folder)
                    _logger.info('Dumping secret value to: %s', output_path)
```

Comment:
if the secret was already dumped because it's backing a cert, why do we still write the secret to the secrets folder? do you see a usecase for that?

Reply:
When you dump the secret you get pfx containing both. I can foresee some cases where a service can directly use that. For example, for Windows based applications.